### PR TITLE
#196 Added an ability to select date and time for the new meeting

### PR DESCRIPTION
**Pull request by Ymkavi**
Screenshot 2022-09-04 173205
Screenshot 2022-09-04 173226

### CODE REVIEW

File: frontend/src/app/modules/new-meeting/new-meeting-calendar/new-meeting-calendar.component.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Input() viewDate: Date;
+
+    @Output() meetingStartTimeChange: EventEmitter<Date> = new EventEmitter<Date>();
+
+    events: CalendarEvent[] = [];
+
+    refresh: Subject<void> = new Subject<void>();
+
+    hourSegmentClicked(date: Date) {
+        if (this.duration.minutes == null) {
```
Comment (tatianahutii) [inline frontend/src/app/modules/new-meeting/new-meeting-calendar/new-meeting-calendar.component.ts]:
```suggestion
        if (!this.duration.minutes) {
```

File: frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts [hunk 2]
```diff
@@ -137,35 +132,37 @@ export class NewMeetingComponent extends BaseComponent implements OnInit {
...
 
-    durationChanged(timeValue: string, unitOfTime: string) {
-        if (unitOfTime === 'hour') {
-            this.convertDuration(timeValue);
+        if (unitOfTime === UnitOfTime.Hour) {
+            customMinutes = parseInt(customTime, 10) * 60;
         } else {
-            this.duration = parseInt(timeValue, 10);
+            customMinutes = parseInt(customTime, 10);
         }
```
Comment (tatianahutii) [inline frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts:157]: you can combine them
Use ternary operator

Comment (tatianahutii) [inline frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts:157]: you can combine them
Use ternary operator

File: frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts [hunk 3]
```diff
@@ -137,35 +132,37 @@ export class NewMeetingComponent extends BaseComponent implements OnInit {
...
+        let customMinutes: number;
 
-    durationChanged(timeValue: string, unitOfTime: string) {
-        if (unitOfTime === 'hour') {
-            this.convertDuration(timeValue);
+        if (unitOfTime === UnitOfTime.Hour) {
+            customMinutes = parseInt(customTime, 10) * 60;
         } else {
-            this.duration = parseInt(timeValue, 10);
+            customMinutes = parseInt(customTime, 10);
```
Comment (tatianahutii) [inline frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts:156]: image
you can combine them
Use ternary operator

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts (hunk 2 above)
Comment (tatianahutii) [inline frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts]: you can combine them
Use ternary operator

**Comment by tatianahutii** [comment]
Could you load availability slots (from settings) and disable not active slot' time?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by hekeemje — APPROVED** [review]
looks good

**Review by apofis96 — APPROVED** [review]

**Pull request merged**